composer.json

```diff
@@ -23,19 +23,19 @@
         }
     },
     "require": {
-        "php": ">=7",
-        "amphp/parallel": "^1.1",
-        "amphp/amp": "^2.0.3",
-        "opis/closure": "^3.0.7"
+        "php": ">=7.4",
```

Comment:
I would highly recommend to switch to `^7.4 || ^8.0` (although some might say you should use `~` and specify the patch version instead)

Reply:
Yeah, I also wanted to use that. 👍🏻 But was just sticking with the existing method for consistency.

Reply:
No, I very much don't agree with that, it's just unnecessary maintenance work.

Reply:
Increased maintenance yes but unnecessary no: you lie about the versions the package supports hence Composer will install incorrect versions requiring the user to dig into the changelogs/history to manually find a suitable tag he/she can use

Reply:
I've spent _a lot_ more time with incorrect upper bounds than I've spent with PHP breaking changes affecting our packages. Composer's recent support for only ignoring upper bounds will improve this, but it's still not worth the time spent IMO.

Reply:
Do you want me to change that back to `>=7.4` then?

Reply:
Please use >=, yes

Reply:
@kelunik, done. 👍🏻

Reply:
Generally to be avoided: PHP 9 is not gonna work OOTB anyway, especially for something as reflection-intensive as `laravel/serializable-closure`.

Should be `^7.4 || ~8.0.0 || ~8.1.0` too, since `8.2.0` is an unknown unknown too.

Reply:
@Ocramius This might be true for `laravel/serializable-closure`, but not for code within this package. I know your opinion there, I just don't agree with it for our packages.

Reply:
That's fine, I'm just generally hopping around packages, explaining why this is objectively wrong, and informing other readers about it.

Reply:
I know. ;-)

Reply:
Sooo... even today it has cost me time: https://github.com/amphp/rpc/commit/dd4a51803869fcccd89bd0c338a8a0d857c6c6d1